### PR TITLE
Removed global oauth variable #5086

### LINKED ISSUE #5086: Isolate OauthMock data for each test
Looks like tests interfere with each other by re-using the same global `oauth` object, possibly overwriting each other's tokens
image

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR removes the global `oauth` variable for better test isolation

close #5086 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Internal test framework change

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by rrrooommmaaa** [comment]
I tucked `oauth` instance into `ConfigurationProvider` class to nicely wire handlers.
However, the same `ConfigurationProvider` object should be used throughout the test to not lose the tokens.
So I modified the tests that use log in provided by `testWithBroswer` until #5076 is fixed.

File: test/source/tests/decrypt.ts [hunk 1]
```diff
@@ -1366,15 +1347,13 @@ XZ8r4OC6sguP/yozWlkG+7dDxsgKQVBENeG6Lw==
     test(
       `verify - sha1 shows error`,
       testWithBrowser('compatibility', async (t, browser) => {
-        t.mockApi!.configProvider = new ConfigurationProvider({
-          attester: {
-            pubkeyLookup: {
-              'sha1@sign.com': {
-                pubkey: sha1signpubkey,
-              },
+        t.mockApi!.configProvider!.config!.attester = {
```
Comment (ioanmo226) [inline test/source/tests/decrypt.ts:1350]: @rrrooommmaaa, do you prefer setting options like this?
I prefer using the `ConfigurationProvider` constructor and passing the config there.

This is because in the near future, we will also have to pass ekm, wkd, and other configs to configProvider, and if 
we don't pass it to constructor we would end up force-unwrapping `configProvider!.config!.objectToConfig` for each config, which is not ideal.

File: test/source/tests/compose.ts [hunk 2]
```diff
@@ -2296,30 +2288,28 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
...
-        t.mockApi!.configProvider = new ConfigurationProvider({
-          attester: {
-            pubkeyLookup: {
-              [recipients.to]: {
-                pubkey: somePubkey,
-              },
-              [recipients.bcc]: {
-                pubkey: somePubkey,
-              },
+        t.mockApi!.configProvider!.config!.attester = {
```
Comment (ioanmo226) [inline test/source/tests/compose.ts:2291]: I propose merging this pull request after #5076.
#5076 is expected to be ready today, and once it is merged, we can revert the changes made in this pull request (https://github.com/FlowCrypt/flowcrypt-browser/pull/5087/commits/37c2fa400800cc1a318cc1dde19432d2919c1026) and close it.

Comment (sosnovsky) [reply test/source/tests/compose.ts:2291]: Sounds reasonable, let's wait for finishing https://github.com/FlowCrypt/flowcrypt-browser/pull/5083, as it part of our bigger test refactoring, then it probably should fix this issue too.

**Review by ioanmo226 — APPROVED** [review]
Looks good on my side. @sosnovsky You may merge PR if you also think it's good.

**Review by sosnovsky — APPROVED** [review]
Good for me too 👍
